server/py/framework/db/sqldb/db.py

```diff
@@ -7149,3 +7152,103 @@ def _paginate_query(
             query = query.limit(limit)
 
         return query
+
+    @staticmethod
+    def _find_artifact_latest_tag(session, object_record):
+        """Find the 'latest' tag for an artifact object."""
+        return (
+            session.query(ArtifactV2.Tag)
+            .filter(
+                ArtifactV2.Tag.obj_id == object_record.id,
+                ArtifactV2.Tag.name == mlrun.common.constants.RESERVED_TAG_NAME_LATEST,
+                ArtifactV2.Tag.project == object_record.project,
+            )
+            .one_or_none()
+        )
+
+    @staticmethod
+    def _find_other_latest_in_other_iterations(session, object_record):
+        """Check if there are other 'latest' tags in other iterations for the same producer_id."""
+        return (
+            session.query(ArtifactV2.Tag)
+            .filter(
+                ArtifactV2.Tag.obj_id != object_record.id,
+                ArtifactV2.Tag.name == mlrun.common.constants.RESERVED_TAG_NAME_LATEST,
+                ArtifactV2.Tag.project == object_record.project,
+                ArtifactV2.Tag.obj_name == object_record.key,
+                ArtifactV2.Tag.obj_id.in_(
+                    session.query(ArtifactV2.id).filter(
+                        ArtifactV2.producer_id == object_record.producer_id,
+                        ArtifactV2.iteration != object_record.iteration,
+                    )
+                ),
+            )
+            .first()
+        )
+
+    @staticmethod
+    def _find_previous_most_recent_artifact(session, object_record):
+        """Find the most recent artifact based on the update timestamp, excluding the current artifact."""
+        query = session.query(ArtifactV2).filter(
+            ArtifactV2.id != object_record.id,
+            ArtifactV2.project == object_record.project,
+            ArtifactV2.key == object_record.key,
+            ArtifactV2.best_iteration,
+        )
+
+        # Return the most recent artifact based on the update timestamp
+        return query.order_by(ArtifactV2.updated.desc()).first()
+
+    def _handle_latest_tag(self, session, object_record, commit):
```

Comment:
```suggestion
    def _handle_latest_tag_deletion(self, session, object_record, commit):
```

Reply:
Also, move this function above all of the `_find_...` functions as it calls them.

Reply:
@TomerShor I think it is more like `_update_artifact_latest_tag`

Reply:
still too general, as this function is a very specific flow that is only relevant to deletion.
I'll meet you in the middle with:
`_update_artifact_latest_tag_on_deletion`